@xen-orchestra/lite/src/components/Console.tsx

```diff
@@ -97,11 +138,11 @@ const Console = withState<State, Props, Effects, Computed, ParentState, ParentEf
               throw new Error('Not connected to XAPI')
             }
 
-            const url = new URL(consoles[0].location)
-            url.protocol = window.location.protocol === 'https:' ? 'wss:' : 'ws:'
-            url.searchParams.set('session_id', xapi.sessionId)
+            this.state.url = new URL(consoles[0].location)
+            this.state.url.protocol = window.location.protocol === Protocols.https ? Protocols.wss : Protocols.ws
```

Comment:
I created an enumerator to avoid missing the `:` after the protocol when we need to test it.
I stayed stuck long time because i was testing `protocol === 'https'` instead of `protocol === 'https:'`.
I think it enhance the readability and avoid mistakes in the futur.